src/logging/Logger.hpp

```diff
@@ -1,15 +1,45 @@
 #pragma once
 
 #include <string>
-#include <boost/log/trivial.hpp>
+#include <memory>
 
 namespace precice {
 namespace logging {
 
-class Logger : public boost::log::sources::severity_logger<boost::log::trivial::severity_level>
-{
+/// Struct used to capture the original location of a log request
+struct LogLocation {
+    const char * file;
+    int line;
+    const char * func;
+};
+
+/// This class provides a leightweight logger.
+class Logger {
 public:
+  /** creates a logger for a given module.
+   * @param[in] the name of the module 
+   */
   explicit Logger(std::string module);
+
+  Logger(const Logger& other);
+  Logger& operator=(Logger other);
```

Comment:
Rule of 5: There should be a move constructor somewhere, right? Why does Logger write its own copy/move anyway? Shouldn't the implicit ones suffice?

Reply:
The default is implicitly deleted as, `std::unique_ptr` has a deleted copy ctor.

Will add the move.